vita3k/shader/src/usse_program_analyzer.cpp

```diff
@@ -150,26 +150,25 @@ bool is_buffer_fetch_or_store(const std::uint64_t inst, int &base, int &cursor,
     return false;
 }
 
-int match_uniform_buffer_with_buffer_size(const SceGxmProgram &program, const SceGxmProgramParameter &parameter, const shader::usse::UniformBufferMap &buffers) {
-    assert(parameter.component_count == 0 && parameter.category == SCE_GXM_PARAMETER_CATEGORY_UNIFORM_BUFFER);
+int get_uniform_buffer_sizes(const SceGxmProgram &program, UniformBufferSizes &sizes) {
+    memset(sizes.data(), 0, sizeof(UniformBufferSizes));
```

Comment:
are you sure if `sizeof` here is correct. It can be true, but it looks strange for me

Reply:
From my points of view, this is correct. UniformBufferSizes is an alias for std::array<std::uint32_t, 15>, so sizeof(UniformBufferSizes) returns 60 and the memset has the expected behavior of setting all the entries of sizes to 0.

Reply:
I see. It just looks strange for me (` memset(&sizes, 0, sizeof(UniformBufferSizes));` or ` memset(sizes.data(), 0, sizes.size()*sizeof(sizes.data()));`)

Reply:
Yes, &sizes points to the same location as sizes.data(), I agree that this can look a bit weird.